gnuradio-runtime/lib/qa_typed_sync_block.cc

```diff
@@ -0,0 +1,116 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2021 Marcus Müller
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ */
+#include <gnuradio/top_block.h>
+#include <gnuradio/typed_sync_block.h>
+#include <boost/test/unit_test.hpp>
+#include <algorithm>
+#include <memory>
+using gr::produced_t;
+
+using super = gr::typed_sync_block<int, float>;
+class int_to_float : public super
+{
+
+public:
+    using sptr = std::shared_ptr<int_to_float>;
+    static sptr make() { return gnuradio::make_block_sptr<int_to_float>(); }
+
+    produced_t
+    work(size_t n_items, inputs_t input_streams, outputs_t output_streams) override
+    {
+        auto& out = output_streams[0];
+        const auto& in = input_streams[0];
+        for (size_t idx = 0; idx < n_items; ++idx) {
+            out[idx] = static_cast<out_type>(in[idx]);
+        }
+        return { n_items };
+    }
+    int_to_float() : super("int_to_float") {}
+};
+
+using int_src = gr::typed_sync_block<char, int>;
+class int_generator : public int_src
+{
+private:
+    size_t _counter = 0;
+
+public:
+    using sptr = std::shared_ptr<int_generator>;
+    static sptr make() { return gnuradio::make_block_sptr<int_generator>(); }
+
+    produced_t
+    work(size_t n_items, inputs_t input_streams, outputs_t output_streams) override
+    {
+        for (const auto& output_stream_span : output_streams) {
+            for (auto& item : output_stream_span) {
+                item = _counter++ & ((1 << 12) - 1);
+            }
+        }
+        return { n_items };
+    }
+    int_generator() : int_src("int_generator", 0, 0) {}
+};
```

Comment:
Just for illustration: this is a simple `int`-producing source, in its entirety.